### PR TITLE
feat(terra-spacer): Add type defs for terra-spacer

### PR DESCRIPTION
**Pull request by juzerzarif**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by juzerzarif** [comment]
Pinging @Berkyle

File: types/terra-spacer/lib/__spacerShorthand.d.ts [hunk 1]
```diff
@@ -0,0 +1,11111 @@
+export type SpacerShorthand =
+  | 'none'
```
Comment (sandersn) [inline types/terra-spacer/lib/__spacerShorthand.d.ts:2]: You could reduce this to about 20 lines of code if you use template literal types from TS 4.1. Unfortunately that would prevent people on older TS versions from using the package, so you'd have to decide if it's worth it.

Comment (juzerzarif) [reply types/terra-spacer/lib/__spacerShorthand.d.ts:2]: I considered that but 4.1 seemed too recent in terms of version support so decided to opt for this mess 😅. @Berkyle do you have an opinion on this?

Comment (Berkyle) [reply types/terra-spacer/lib/__spacerShorthand.d.ts:2]: Probably best to be backwards compatible with 3.x, I don't think we'll be able to enforce 4.x or higher across terra. This is a good note, though

**Pull request merged**